### DISCUSSION TITLE
deploying to github pages

### DISCUSSION THREAD

**Discussion by coxauto-kennyg** [observablehq/framework#1052] [Show and tell]
Hi! I just wanted to drop a note about my experience publishing to GitHub Pages on an internal GitHub enterprise instance. to document in case someone else ran into the same issues. It's not really a bug, but I did have to piece things together things that I couldn't find documented.

I had to make a couple of changes to get things to work. I created a `gh-pages` branch for publishing the source. I updated the config to use a different directories since GitHub Pages only allows publishing pages from the `root` or `docs` directory. I also added a `.nojekyll` file in the `docs` directory, otherwise the javascript failed to load, reference https://docs.github.com/en/pages/getting-started-with-github-pages/about-github-pages#static-site-generators.

**Comment by Fil** [comment]
Thanks for sharing. `.nojekyll` does the trick by allowing folders that begin with a `_` to be deployed (there are certainly other ways to deploy on github pages that do not need this hack). I'm moving this to a discussion since it's not a bug of Framework.

**Comment by llimllib** [comment]
I [wrote a note](https://notes.billmill.org/programming/observable_framework/github_workflow_for_publishing_an_observable_framework.html) that documents how you can use a github action to publish to github pages instead of the `gh-pages` branch, you might find it useful, as It avoids any jekyll nonsense.
